cloud-core/src/main/java/cloud/commandframework/exceptions/CommandExecutionException.java

```diff
@@ -0,0 +1,55 @@
+//
+// MIT License
+//
+// Copyright (c) 2020 Alexander Söderberg & Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+package cloud.commandframework.exceptions;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Exception thrown when there is an exception during execution of a command handler
+ * @since 1.2.0
+ */
+public class CommandExecutionException extends IllegalArgumentException {
+
+    private static final long serialVersionUID = -4785446899438294661L;
+
+    private final Throwable cause;
+
+    /**
+     * Exception thrown when there is an exception during execution of a command handler
+     *
+     * @param cause cause
+     */
+    public CommandExecutionException(final @NonNull Throwable cause) {
```

Comment:
```suggestion
    public CommandExecutionException(final @NonNull Exception cause) {
```
Errors shouldn't be caught.